src/org/rust/lang/RustDescriptionProvider.kt

```diff
@@ -0,0 +1,52 @@
+package org.rust.lang
+
+import com.intellij.codeInsight.highlighting.HighlightUsagesDescriptionLocation
+import com.intellij.psi.ElementDescriptionLocation
+import com.intellij.psi.ElementDescriptionProvider
+import com.intellij.psi.PsiElement
+import com.intellij.usageView.UsageViewLongNameLocation
+import com.intellij.usageView.UsageViewNodeTextLocation
+import com.intellij.usageView.UsageViewShortNameLocation
+import com.intellij.usageView.UsageViewTypeLocation
+import org.rust.lang.core.psi.RustPatBinding
+
+class RustDescriptionProvider : ElementDescriptionProvider {
+    override fun getElementDescription(element: PsiElement, location: ElementDescriptionLocation): String? {
+        return when (location) {
+            is UsageViewNodeTextLocation -> getNodeText(element)
+            is UsageViewShortNameLocation -> getShortName(element)
+            is UsageViewLongNameLocation -> getLongName(element)
+            is UsageViewTypeLocation -> getType(element)
+            is HighlightUsagesDescriptionLocation -> getLongName(element)
+            else -> null
```

Comment:
We are trying to keep arrow aligned in `when`. There is a checkbox for it in code style config.